httpserver/handlers.go

```diff
@@ -58,35 +60,46 @@ func (server *Server) verify(ctx context.Context, w http.ResponseWriter, r *http
 		wg.Add(1)
 		go func(subject string) {
 			defer wg.Done()
-			// TODO: Enable caching:  Providers should add a caching mechanism to avoid extra calls to external data sources.
-			logrus.Infof("subject for request %v %v is %v", sanitizedMethod, sanitizedURL, subject)
-			returnItem := externaldata.Item{
-				Key: subject,
-			}
+			unlock := server.keyMutex.Lock(subject)
+			defer unlock()
 
-			verifyParameters := e.VerifyParameters{
-				Subject: subject,
-			}
-
-			result, err := server.GetExecutor().VerifySubject(ctx, verifyParameters)
-			if err != nil {
-				returnItem.Error = err.Error()
+			logrus.Infof("subject for request %v %v is %v", sanitizedMethod, sanitizedURL, subject)
+			var result types.VerifyResult
+			res := server.cache.get(subject)
+			if res != nil {
+				logrus.Infof("cache hit for subject %v", subject)
```

Comment:
I think we may want to make these debug logs

Reply:
for sure! I had them enabled so we can see it in the logs for CI to confirm it's working as expected. I'll change it to `Debugf`.

Reply:
done!